### ISSUE TITLE
Cache keys in ITokenValidableExtension.AreTokensValidAsync

### ISSUE THREAD

**Issue by meum** [microsoftgraph/msgraph-sdk-dotnet-core#948]
### Is your feature request related to a problem? Please describe the problem.

Every time I call AreTokensValidAsync, I see two requests to login.microsoftonline.com. One to `/common/.well-known/openid-configuration` and one to `/common/discovery/keys`.
I assume these keys don't change very often, so they should be cached.

### Describe the solution you'd like.

I would like the keys to be cached, to avoid these two requests being repeated so often.

**Comment by andrueastman** [comment]
Thanks for raising this @meum 

Taking a look at this, this looks to be caused by the call here to `configurationManager.GetConfigurationAsync()`

https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/fa698f16386655667a0ad54dbea318112301a123/src/Microsoft.Graph.Core/Extensions/ITokenValidableExtension.cs#L70C17-L70C30

Arguably if the same `wellKnownUri` parameter is used, it would be useful to cache this property to avoid repeated fetching of configuration whenever the function is called. 

Any chance this is something you'd be willing to submit a PR for?

**Comment by meum** [comment]
Yes, I can try to make a PR. Is there caching elsewhere in Microsoft.Graph.Core I can look at?
